src/test/run-pass/typeid-intrinsic.rs

```diff
@@ -91,4 +92,11 @@ pub fn main() {
     // Check fn pointer against collisions
     assert!(TypeId::of::<fn(fn(A) -> A) -> A>() !=
             TypeId::of::<fn(fn() -> A, A) -> A>());
+
+    // Check const
+    const T1: TypeId = TypeId::of::<A>();
```

Comment:
This should be in a separate file to ensure that `TypeId::of` without the feature enabled still works

Reply:
I agree, this test is only temporal as I didn't know what to test. (It still works btw).